specification/trafficmanager/resource-manager/Microsoft.Network/stable/2020-12-01/trafficmanager.json

```diff
@@ -0,0 +1,1659 @@
+        },
+        "message": {
+          "type": "string",
+          "description": "Descriptive message that explains why the name is not available, when applicable."
+        }
+      },
+      "description": "Class representing a Traffic Manager Name Availability response."
+    },
+    "Region": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The code of the region"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the region"
+        },
+        "regions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Region"
+          },
+          "description": "The list of Regions grouped under this Region in the Geographic Hierarchy."
+        }
+      },
+      "description": "Class representing a region in the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "GeographicHierarchyProperties": {
+      "properties": {
+        "geographicHierarchy": {
+          "$ref": "#/definitions/Region",
+          "description": "The region at the root of the hierarchy from all the regions in the hierarchy can be retrieved."
+        }
+      },
+      "description": "Class representing the properties of the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "TrafficManagerGeographicHierarchy": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GeographicHierarchyProperties",
+          "description": "The properties of the Geographic Hierarchy resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Class representing the Geographic hierarchy used with the Geographic traffic routing method."
+    },
+    "UserMetricsProperties": {
+      "properties": {
+        "key": {
+          "type": "string",
+          "description": "The key returned by the User Metrics operation."
+        }
+      },
+      "description": "Class representing a Traffic Manager Real User Metrics key response."
+    },
+    "Resource": {
+      "description": "The core properties of ARM resources",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Fully qualified resource Id for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/trafficManagerProfiles/{resourceName}"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource"
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the resource. Ex- Microsoft.Network/trafficManagerProfiles."
+        },
+        "systemData": {
+          "$ref": "#/definitions/SystemData",
```

Comment:
What's the motivation behind using a custom reference instead of using the one from common-types?

Reply:
We already have copies from common-types in trafficmanager.json that vary because the read-only flag is removed, and some exceptions to allow it. When I tried to use common-types for system data I got errors; it seems like the validation tools encounter ambiguity when the .json file has these copies and also references common-types.

Reply:
I tried a lot of variations in a different PR (https://github.com/Azure/azure-rest-api-specs/pull/12249) to get around this. I think that ultimately we have to have different objects for our requests and responses, so that we don't have to make an exception for the readonly flag. But I'm trying to use the lightest touch here to make a new version for a trivial change. Fixing the request / response problem would be a different PR.

Reply:
Ok, is not a blocker, was more a wondering. It is ok to copy the definition many teams do the same but I pointed it out in case you were not familiar with common-types.